src/vse_sync_pp/analyzers/ppsdpll.py

```diff
@@ -0,0 +1,209 @@
+### SPDX-License-Identifier: GPL-2.0-or-later
+
+import pandas as pd
+import numpy as np
+import allantools
+
+from collections import namedtuple
+from decimal import Decimal
+
+class Analyzer():
+    def __init__(self):
+        self._col1_name = 'timestamp'
+        self._col2_name = 'eecstate'
+        self._col3_name = 'phasestate'
+        self._col4_name = 'phaseoffset'
+        self._data  = pd.DataFrame(columns=['timestamp','eecstate','phasestate','phaseoffset']) 
+    def append(self, data):
+        """Append x and y data points extracted from `data`"""
+        col1_val = getattr(data, self._col1_name)
+        col2_val = getattr(data, self._col2_name)
+        col3_val = getattr(data, self._col3_name)
+        col4_val = getattr(data, self._col4_name)
+        self._data.loc[len(self._data)]=[col1_val,col2_val,col3_val,float(col4_val)]
+    def retrieve(self):
+        """Return the data in Data Frame format"""
+        return self._data
+    def to_visualize_stage(self, col1, col2, col3, col4):
+        elems = (self._col1_name, self._col2_name, self._col3_name, self._col4_name)
+        data = namedtuple('Analyzed', elems)
+        timestamp = Decimal(col1)
+        eecstate = int(col2)
+        phasestate = int(col3)
+        phaseoffset = float(col4)
+        return data(timestamp, eecstate, phasestate, phaseoffset)
+    def transform(self):
+        """Generator to convert from Data Frame to NamedTuple (canonical)"""
+        result = [self.to_visualize_stage(row[0],row[1],row[2],row[3]) for row in zip(self._data['timestamp'],self._data['eecstate'],self._data['phasestate'],self._data['phaseoffset'])]
```

Comment:
You should probably use `self._data.itertuples()` 
https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.itertuples.html#pandas.DataFrame.itertuples

Reply:
Assuming that the name tuple is in the correct order

Reply:
transform and to_visualize_stage improved and simplified in this method:
```
def format_to_canonical(self):
        """Generator to convert from Data Frame to an accepted Analyzed canonical format"""
        yield from self._data.itertuples(index=False, name='Analyzed')
```